### ISSUE TITLE
Feature: add `partition-by-new-line` option to rules

### ISSUE THREAD

**Issue by Azarattum** [azat-io/eslint-plugin-perfectionist#86]
### What rule do you want to change?

All the multiline rules

### Describe the problem

We often want to group things together by some business logic that in hard to describe as a rule (or there isn't such a rule yet). Instead of implementing a rule for each use case, you can allow users to decide on their grouping on case by case basis. The simplest way to allow that is make an option to treat new line as a group delimiter.

### Code example

You might want to have a type that is sorted like this (separating data, state and methods):
```ts
type Page<T> = {
  pages: ReturnType<typeof page>[];
  items: T[];
  
  loaded: Promise<void>;
  completed: boolean;
  progress: number;
  number: number;

  next(): boolean;
  prev(): boolean;
  close(): void;
};
```

Or you might want to have getters sorted separately in your objects (but such rule exists only for classes):
```ts
return {
  satisfied(id: number) {
    return progress[id].size >= size;
  },
  has(item: ToInfo<T> | T) {
    return items.has(item);
  },

  get loaded() {
    if (this.progress >= 1) return Promise.resolve();
    return wait();
  },
  get items() {
    return items.items;
  },
}
```

And so on...

### Validations

- [X] Read the [docs](https://eslint-plugin-perfectionist.azat.io).
- [X] Check that there isn't [already an issue](https://github.com/azat-io/eslint-plugin-perfectionist/issues) that reports the same bug to avoid creating a duplicate.

**Comment by azat-io** [comment]
That's a good idea. What rules do you think need this feature? I think the following:

- sort-interfaces
- sort-object-types
- sort-objects

**Comment by azat-io** [comment]
Added in 8654e44138f16d17ecafefbce21efbe70a7ce9b0, 48532ef344240c66ed9f92be74beb3d3b31ca531 and 48532ef344240c66ed9f92be74beb3d3b31ca531.
Released in [v2.3.0](https://github.com/azat-io/eslint-plugin-perfectionist/releases/tag/v2.3.0).
